src/handler/archive.ts

```diff
@@ -24,8 +24,8 @@ export const handler = async (event: APIGatewayProxyEvent): Promise<APIGatewayPr
 
     const body: ArchiveRecordRequestBody = JSON.parse(event.body as string) as ArchiveRecordRequestBody;
 
-    const systemNumber = event?.pathParameters?.systemNumber ?? '';
-    const createdTimestamp = event?.pathParameters?.createdTimestamp ?? '';
+    const systemNumber = decodeURIComponent(event?.pathParameters?.systemNumber ?? '');
+    const createdTimestamp = decodeURIComponent(event?.pathParameters?.createdTimestamp ?? '');
     const userDetails = getUserDetails(event.headers.Authorization ?? '');
 
```

Comment:
Should systemNumber and createdTimestamp always be provided when trying to archive a technical-record, to ensure the correct unique one is identified?

Not necessarily saying it's a change for you to make here, but the existing code implies that they may not be present, so are set to `''`, making the subsequent call to `getBySystemNumberAndCreatedTimestamp` on the database obsolete?

I'll make a note of this and check the previous tickets to see if there was a reason for it.

Reply:
They are validated in a method just above as being there, but the compiler will not realise it. We opted for this instead of a ! for lining reasons but essentially the ?? path is non-reachable code we are using to keep the compiler happy. Sucks, but if you look, basically every endpoint has this.

Reply:
Okay fair enough, maybe we can revisit this across the services at some point. Approved.